src/main/scala/encry/network/peer/PeerManager.scala

```diff
@@ -50,9 +50,8 @@ class PeerManager extends Actor with Logging {
     case Handshaked(peer) =>
       if (peer.direction == Outgoing && isSelf(peer.socketAddress, peer.handshake.declaredAddress))
         peer.handlerRef ! CloseConnection
-      else if (checkPossibilityToAddPeer(peer.socketAddress)) {
-        if (peer.publicPeer) self ! AddOrUpdatePeer(peer.socketAddress, Some(peer.handshake.nodeName), Some(peer.direction))
-        else PeerDatabase.remove(peer.socketAddress)
+      else if (checkPossibilityToAddPeer(peer.socketAddress) && !connectedPeers.contains(peer.socketAddress)) {
```

Comment:
We call `checkPossibilityToAddPeer` on both sending and receiving of AddOrUpdatePeer msg, I think one check is enough